### PR TITLE
mgid bid adapters: refactoring for trimmer code

### PR DESCRIPTION
**Pull request by sanychtasher**
## Type of change
- [ ] Updated bidder adapter 
- [ ] Refactoring (no functional changes, no api changes)

## Description of change
Deduplicate code in mgid prebid adapter

### CODE REVIEW

File: modules/mgidXBidAdapter.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { BANNER, NATIVE, VIDEO } from '../src/mediaTypes.js';
 import { config } from '../src/config.js';
 import { USERSYNC_DEFAULT_CONFIG } from '../src/userSync.js';
 import { isBidRequestValid, buildRequestsBase, interpretResponse } from '../libraries/teqblazeUtils/bidderUtils.js';
+import { getUserSyncs } from './mgidBidAdapter.js'
```
Comment (patmmccann) [inline modules/mgidXBidAdapter.js:8]: Generally this is looking great! However, you can import from /libraries/..., not from another adapter, so please move this function to a file in that folder for both

**Comment by patmmccann** [comment]
I see you're triggering warnings when you try to move stuff out of the redtram adapter. This is very much appreciated and you don't have to solve all of redtram for your mgid changes to merge in

Thanks!!!

**Comment by patmmccann** [comment]
Apologies if my last comment was misleading, we really appreciate the change to redtram and undertone, I was just saying you can ignore the robot complaining about other problems on those other adapters

**Comment by sanychtasher** [comment]
@patmmccann We don't know how to fix it more correctly 
[modules/adtrueBidAdapter.js](https://github.com/prebid/Prebid.js/blob/60108e6113d5701a008acd6810044eae7880353e/modules/adtrueBidAdapter.js#L50-L55) has 8 duplicated lines with [modules/mgidBidAdapter.js](https://github.com/prebid/Prebid.js/blob/60108e6113d5701a008acd6810044eae7880353e/modules/mgidBidAdapter.js#L39-L43).
Maybe you could recommend a solution

**Comment by patmmccann** [comment]
> @patmmccann We don't know how to fix it more correctly [modules/adtrueBidAdapter.js](https://github.com/prebid/Prebid.js/blob/60108e6113d5701a008acd6810044eae7880353e/modules/adtrueBidAdapter.js#L50-L55) has 8 duplicated lines with [modules/mgidBidAdapter.js](https://github.com/prebid/Prebid.js/blob/60108e6113d5701a008acd6810044eae7880353e/modules/mgidBidAdapter.js#L39-L43). Maybe you could recommend a solution
i

I would recommend not worrying about that in this pr and covering later.

**Comment by sanychtasher** [comment]
@patmmccann Can you please check the PR? Thanks!

**Review by patmmccann — APPROVED** [review]

**Pull request merged**